### PR TITLE
OKTA-489915 Updating flow diagram and text in the OAuth 2.0 and OpenID Connect Overview

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Updated flow diagram and some following text (including brief content on Interaction Code flow) in the [OAuth 2.0 and OpenID Connect Overview ](https://developer.okta.com/docs/concepts/oauth-openid/#is-your-app-high-trust)concept doc.
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-489915](https://oktainc.atlassian.net/browse/OKTA-489915)

### CODE REVIEW

**Review by aarongranick-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/concepts/oauth-openid/index.md [hunk 1]
```diff
@@ -103,9 +103,7 @@ A client application is considered public when an end user could possibly view a
 
 #### Is your client a SPA or native?
 
-If your client application is a SPA you should use the [Implicit flow](#implicit-flow).
-
-If your client application is a native application, you should use the [Authorization Code flow with PKCE](#authorization-code-flow-with-pkce).
+If your client application is a SPA or native application, you should use an authorization flow with PKCE, either the [Interaction Code flow with PKCE](#interaction-code-flow) or [Authorization Code flow with PKCE](#authorization-code-flow-with-pkce). If you are doing a redirect flow to an Okta hosted sign-in page, the authorization code flow with PKCE is recommended. If you want to embed the sign-in experience, the interaction code flow is recommended.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/concepts/oauth-openid/index.md:106]:
```suggestion
If your client application is a SPA or a native application, you should use an authorization flow with PKCE, such as either the [Interaction Code flow with PKCE](#interaction-code-flow) or the [Authorization Code flow with PKCE](#authorization-code-flow-with-pkce). If you are doing a redirect flow to an Okta-hosted sign-in page, the Authorization Code flow with PKCE is recommended. If you want to embed the sign-in experience, the Interaction Code flow is recommended.
```
I was told to capitalize the flow names.

File: packages/@okta/vuepress-site/docs/concepts/oauth-openid/index.md [hunk 2]
```diff
@@ -117,6 +115,41 @@ If you own both the client application and the resource that it's accessing, the
 
 If your app is not high-trust, you should use the [Authorization Code](/docs/guides/implement-grant-type/authcode/main/) flow.
 
+### Interaction Code flow
+
+The interaction code flow requires clients to pass a client ID, as well as a Proof Key for Code Exchange (PKCE), to keep the flow secure. The user can start the request with minimal information, relying on the client to facilitate the interactions with the Identity Engine component of the Okta Authorization Server to progressively authenticate the user. See [Interaction Code grant type](/docs/concepts/interaction-code/).
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/concepts/oauth-openid/index.md:120]:
```suggestion
The Interaction Code flow requires clients to pass a client ID, as well as a Proof Key for Code Exchange (PKCE), to keep the flow secure. The user can start the request with minimal information, relying on the client to facilitate the interactions with the Identity Engine component of the Okta Authorization Server to progressively authenticate the user. See [Interaction Code grant type](/docs/concepts/interaction-code/).
```

**Review by vanngo-okta — APPROVED** [review]
Looks good! A few nits - approving to not hold you back.

File: packages/@okta/vuepress-site/docs/concepts/oauth-openid/index.md (hunk 1 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/oauth-openid/index.md:106]: Thanks!

**Pull request merged**